src/lucky/base_http_client.cr

```diff
@@ -1,28 +1,100 @@
 require "http/client"
 
-class Lucky::BaseHTTPClient
+# A client for making HTTP requests
+#
+# Makes it easy to pass params, use Lucky route helpers, and chain header methods.
+abstract class Lucky::BaseHTTPClient
+  private getter client
+
   @client : HTTP::Client
 
   def initialize(host = Lucky::Server.settings.host, port = Lucky::Server.settings.port)
     @client = HTTP::Client.new(host, port: port)
   end
 
-  def get(path : String, headers : HTTP::Headers? = nil, params : HTTP::Params? = nil)
-    if params
-      path = path + "?#{params}"
+  {% for method in [:get, :put, :patch, :post] %}
+    def self.{{ method.id }}(*args, **named_args)
+      new.{{ method.id }}(*args, **named_args)
     end
-    @client.get(path, headers: headers)
-  end
+  {% end %}
 
-  {% for method in [:put, :patch, :post] %}
+  # Set headers for requests
+  #
+  # ```
+  # # `content_type` will be normalized to `content-type`
```

Comment:
Headers are case insensitive! https://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive So  technically it does  not capitalize them, but it does change the underscore to a hyphen. That's done  by Crystal :D 

Should  I add a note that headers are case insensitive to the docs?

Reply:
However it doesn't appear to change the underscore to a hyphen:

```
HTTP::Headers{"Foo" => "bar",
 "accept" => "text/csv",
 "accept-encoding" => "gzip, deflate",
 "content-length" => "2",
 "content_type" => "application/json",
 "host" => "localhost:6226",
 "user-agent" => "Crystal"}
```

I think most implementations of header parsers convert underscore to  hyphen  (like Crystal's does) but I think we should do it before setting just in case :)

https://github.com/crystal-lang/crystal/blob/5e6a1b672a8b06d2a819210b9905b806ec3d7c71/src/http/headers.cr#L3-L4